dev/how-to/troubleshoot/tidb-in-kubernetes.md

```diff
@@ -0,0 +1,268 @@
+
+TiDB Operator uses PV (Persistent Volume) and PVC (Persistent Volume Claim) to store persistent data. If you accidentally delete the cluster using `helm delete`, the PV/PVC objects and data are still retained to ensure data security.
+
+To restore the cluster at this time, use the `helm install` command to create a cluster with the same name. The retained PV/PVC and data are reused.
+
+{{< copyable "shell-regular" >}}
+
+```shell
+helm install pingcap/tidb-cluster -n ${releaseName} --namespace=${namespace} --version=v1.0.0-beta.3 -f values.yaml
+```
+
+## Pod is not created normally
+
+After creating a cluster using `helm install`, if the Pod is not created, you can diagnose it using the following commands:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get tidbclusters -n ${namespace}
+kubectl get statefulsets -n ${namespace}
+kubectl describe statefulsets -n ${namespace} ${cluster-name}-pd
+```
+
+## The network connection issue between Pods
+
+In a TiDB cluster, you can access most Pods by using the Pod's domain name (allocated by the Headless Service). The exception is when TiDB Operator collects the cluster information or issues the control command, it accesses the PD (Placement Driver) cluster using the `ServiceName` of the PD service.
+
+When you find some network connection issue exists between Pods from the log or monitoring metrics, or you find the network connection between Pods might be abnormal according to the problem condition, you can follow the following process to diagnose and gradually narrow down the problem:
+
+1. Confirm that the endpoints of the Service and Headless Service are normal:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl -n ${namespace} get endpoints ${cluster_name}-pd
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tidb
+    kubectl -n ${namespace} get endpoints ${cluster_name}-pd-peer
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tikv-peer
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tidb-peer
+    ```
+
+    The `ENDPOINTS` field shown in the above command should be a comma-separated list of `cluster_ip:port`. If the field is empty or incorrect, check the health of the Pod and whether `kube-controller-manager` is working properly.
+
+2. Enter the Pod's Network Namespace to diagnose network problems:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tkctl debug -n ${namespace} ${pod_name}
+    ```
+
+    After the remote shell is started, use the `dig` command to diagnose the DNS resolution. If the DNS resolution is abnormal, refer to [Debugging DNS Resolution](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/) for troubleshooting.
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    dig ${HOSTNAME}
+    ```
+
+    Use the `ping` command to diagnose whether the three layers of network to the destination IP are connected (the destination IP is the ClusterIP resolved using `dig`):
```

Comment:
```suggestion
    Use the `ping` command to diagnose the connection with the destination IP (the ClusterIP resolved using `dig`):
```

Reply:
Ping command sends ICMP packets which should belong to the layer 3 network, but I don't think we need to mention it.